assignment1/src/main/java/ro/utcn/sd/he/assignment1/controller/CommandLineController.java

```diff
@@ -86,24 +89,160 @@ private boolean handleCommand(String command){
                 }
                 return false;
 
+            case "listAnswers":
+                handleListAnswers();
+                return false;
+
+            case "deleteAnswer":
+                handleDeleteAnswer();
+                return false;
+
+            case "edit":
+                handleEdit();
+                return false;
+
+            case "register":
+                handleRegister();
+                return false;
+
+            case "show":
+                handleShow();
+                return false;
+
+            case "answer":
+                handleAnswer();
+                return false;
             default:
                 System.out.println("Command unknwon");
                 return false;
         }
 
     }
 
+    private void handleListAnswers(){
+        answerService.findAllAnswers().forEach( a -> print(a.toString()));
+    }
+
+    private void handleEdit(){
+        if(isLogged()){
+            print("What answer do you want to edit (id):");
+            int answerId = scanner.nextInt();
+            scanner.nextLine(); //dump remaining line
+            Answer answer = answerService.findById(answerId);
+            if(userService.getAuthorOf(answer).getUsername().equals(user.getUsername())){
+                print("New text:");
+                String text = scanner.nextLine();
+                answer.setText(text);           //TODO when it is edited, should i change the creation time? or add last edited time?
+                answerService.saveAnswer(answer);
+            } else {
+                print("You can edit only your own questions!");
+            }
+        } else {
+            print("You need to be logged in to edit an answer!");
+        }
+    }
+
+
+    private void handleRegister(){
+        print("enter the username you want to login with:");
+        String username = scanner.next().trim();
+        scanner.nextLine();
+        print("enter a stronk password:");
+        String password = scanner.next().trim();
+        scanner.nextLine();
+        User user = new User(0,username,password,"user",false,0);
+        user = userService.saveUser(user);
```

Comment:
Same as above; you should just pass the inputs (username, password) to the service layer and instantiate + save the user from that layer.

Otherwise, when you will replace this controller with a REST / WebSocket controller, then you will have to migrate / re-make this logic.